### PR TITLE
Register template

### PR DESCRIPTION
**Pull request by CyclingNinja**
Registers the pubpackage with the sunpy packaing-template repo

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[run]
+omit =
+  sunpy-soar/conftest.py
+  sunpy-soar/*setup_package*
+  sunpy-soar/extern/*
+  sunpy-soar/version*
+  */sunpy-soar/conftest.py
+  */sunpy-soar/*setup_package*
+  */sunpy-soar/extern/*
+  */sunpy-soar/version*
```
Comment (nabobalis) [inline .coveragerc:10]:
```suggestion
  sunpy_soar/conftest.py
  sunpy_soar/*setup_package*
  sunpy_soar/extern/*
  sunpy_soar/version*
  */sunpy_soar/conftest.py
  */sunpy_soar/*setup_package*
  */sunpy_soar/extern/*
  */sunpy_soar/version*
```

File: .cruft.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "template": "https://github.com/sunpy/package-template",
+  "commit": "465654fc0c50e79eaca59b629128b56982e0c4e2",
+  "checkout": null,
+  "context": {
+    "cookiecutter": {
+      "package_name": "sunpy-soar",
+      "module_name": "sunpy-soar",
```
Comment (nabobalis) [inline .cruft.json:8]:
```suggestion
      "module_name": "sunpy_soar",
```

File: .cruft.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "short_description": "A sunpy FIDO plugin for accessing data in the Solar Orbiter Archive (SOAR).",
+      "author_name": "The SunPy Community",
+      "author_email": "sunpy@googlegroups.com",
+      "project_url": "https://sunpy.org",
+      "license": "BSD 2-Clause",
+      "minimum_python_version": "3.9",
```
Comment (nabobalis) [inline .cruft.json:14]:
```suggestion
      "minimum_python_version": "3.10",
```
Time to bump this

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (nabobalis) [inline .github/workflows/ci.yml:1]:
```suggestion
```

File: .isort.cfg [hunk 5]
```diff
@@ -0,0 +1,16 @@
+[settings]
+balanced_wrapping = true
+skip =
+    docs/conf.py
+    sunpy-soar/__init__.py
```
Comment (nabobalis) [inline .isort.cfg:5]:
```suggestion
    sunpy_soar/__init__.py
```

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (nabobalis) [inline .pre-commit-config.yaml:1]:
```suggestion
```

File: .ruff.toml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+target-version = "py310"
+line-length = 110
+exclude = [
+  ".git,",
+  "__pycache__",
+  "build",
+  "sunpy-soar/version.py",
```
Comment (nabobalis) [inline .ruff.toml:7]:
```suggestion
  "sunpy_soar/version.py",
```

File: docs/conf.py [hunk 8]
```diff
@@ -1,4 +1,10 @@
 # Configuration file for the Sphinx documentation builder.
+#
+# This file does only contain a selection of the most common options. For a
+# full list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
+
+import datetime
```
Comment (nabobalis) [inline docs/conf.py:7]:
```suggestion
```

File: pyproject.toml [hunk 9]
```diff
@@ -1,31 +1,47 @@
...
+zip-safe = false
+include-package-data = true
+
+[tool.setuptools.packages.find]
+
+[tool.setuptools_scm]
+write_to = "ndcube/_version.py"
+
+
+
```
Comment (nabobalis) [inline pyproject.toml:44]: The lines below here, got to go, they were part of ruff.

File: pytest.ini [hunk 10]
```diff
@@ -0,0 +1,27 @@
+[pytest]
+minversion = 7.0
+testpaths =
+    sunpy-soar
```
Comment (nabobalis) [inline pytest.ini:4]:
```suggestion
    sunpy_soar
```

File: tox.ini [hunk 11]
```diff
@@ -1,46 +1,85 @@
...
+
+# Run the tests in a temporary directory to make sure that we don't import
+# the package from the source tree
+change_dir = .tmp/{envname}
 description =
     run tests
+    oldestdeps: with the oldest supported version of key dependencies
     devdeps: with the latest developer version of key dependencies
-setenv =
+
```
Comment (nabobalis) [inline tox.ini:22]:
```suggestion
```

File: tox.ini [hunk 12]
```diff
@@ -1,46 +1,85 @@
...
+    # If the user has set a LC override we should follow it
+    LC_ALL
+
+set_env =
     MPLBACKEND = agg
     COLUMNS = 180
+    devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+    # Define the base test command here to allow us to add more flags for each tox factor
     PYTEST_COMMAND = pytest -vvv -s -raR --pyargs sunpy_soar --cov-report=xml --cov=sunpy_soar {toxinidir}/docs
+
```
Comment (nabobalis) [inline tox.ini:39]:
```suggestion
```

File: tox.ini [hunk 13]
```diff
@@ -1,46 +1,85 @@
...
+    # devdeps: git+https://github.com/ndcube/ndcube
+    oldestdeps: minimum_dependencies
     pytest-cov
     devdeps: git+https://github.com/sunpy/sunpy
+    pytest-xdist
+# The following indicates which extras_require will be installed
 extras =
     tests
-commands =
+
```
Comment (nabobalis) [inline tox.ini:52]:
```suggestion
```

File: tox.ini (hunk 11 above)
Comment (Cadair) [reply tox.ini:22]: line break removals should be done in the package template.

File: .cruft.json (hunk 3 above)
Comment (CyclingNinja) [reply .cruft.json:14]: will do this, but needs to be done with cruft locally

**Review by nabobalis — APPROVED** [review]

**Pull request merged**